### ISSUE TITLE
simple example doesn't work with Chrome/Firefox in https

### ISSUE THREAD

**Issue by djdeath** [h2o/h2o#1595]
With compiling examples/libh2o/simple.c with USE_HTTP 1, I cannot connect to it with Firefox, nor Chrome.
This is the error I get from Firefox : 
```
Your connection is not secure

The website tried to negotiate an inadequate level of security.

localhost uses security technology that is outdated and vulnerable to attack. An attacker could easily reveal information which you thought to be safe. The website administrator will need to fix the server first before you can visit the site.

Error code: NS_ERROR_NET_INADEQUATE_SECURITY
```

**Comment by concatime** [comment]
This has to do with https inside localhost. Total speculation: I think you need a valid certificate first, but this is not really possible since you have no domain.

**Comment by djdeath** [comment]
Thanks, but in both firefox & chrome you can bypass the "Your connection is not secure" by clicking "Advanced" then "Add Exception". With another webserver (using libsoup) and the same certificate, it works :/

**Comment by Varbin** [comment]
@djdeath This does not seem to be a certificate error. I might be that h2o offers some cipher suites (I personally think there should be a better default...) that are not allowed with HTTP/2. So Firefox and Chrome disallow a connection. 

See also #1331 .

**Comment by Varbin** [comment]
Additionally ECDHE suites need to be turned on for the simple example. Patch:

```diff
--- a/examples/libh2o/simple.c
+++ b/examples/libh2o/simple.c
@@ -32,9 +32,10 @@
 #include "h2o/http2.h"
 #include "h2o/memcached.h"
 
-#define USE_HTTPS 0
+#define USE_HTTPS 1
 #define USE_MEMCACHED 0
 
+
 static h2o_pathconf_t *register_handler(h2o_hostconf_t *hostconf, const char *path, int (*on_req)(h2o_handler_t *, h2o_req_t *))
 {
     h2o_pathconf_t *pathconf = h2o_config_register_path(hostconf, path, 0);
@@ -179,7 +180,8 @@ static int create_listener(void)
 
 #endif
 
-static int setup_ssl(const char *cert_file, const char *key_file)
+
+static int setup_ssl(const char *cert_file, const char *key_file, const char *ciphers)
 {
     SSL_load_error_strings();
     SSL_library_init();
@@ -194,6 +196,7 @@ static int setup_ssl(const char *cert_file, const char *key_file)
                                                   86400);
         h2o_socket_ssl_async_resumption_setup_ctx(accept_ctx.ssl_ctx);
     }
+    SSL_CTX_set_ecdh_auto(accept_ctx.ssl_ctx, 1);
 
     /* load certificate and private key */
     if (SSL_CTX_use_certificate_file(accept_ctx.ssl_ctx, cert_file, SSL_FILETYPE_PEM) != 1) {
@@ -205,6 +208,11 @@ static int setup_ssl(const char *cert_file, const char *key_file)
         return -1;
     }
 
+    if (SSL_CTX_set_cipher_list(accept_ctx.ssl_ctx, ciphers) != 1) {
+        fprintf(stderr, "ciphers could not be set");
+        return -1;
+    }
+
 /* setup protocol negotiation methods */
 #if H2O_USE_NPN
     h2o_ssl_register_npn_protocols(accept_ctx.ssl_ctx, h2o_http2_npn_protocols);
@@ -239,7 +247,7 @@ int main(int argc, char **argv)
     if (USE_MEMCACHED)
         h2o_multithread_register_receiver(ctx.queue, &libmemcached_receiver, h2o_memcached_receiver);
 
-    if (USE_HTTPS && setup_ssl("examples/h2o/server.crt", "examples/h2o/server.key") != 0)
+    if (USE_HTTPS && setup_ssl("examples/h2o/server.crt", "examples/h2o/server.key", "DEFAULT:!MD5:!DSS:!DES:!RC4:!RC2:!SEED:!IDEA:!NULL:!ADH:!EXP:!SRP:!PSK") != 0)
         goto Error;
 
     /* disabled by default: uncomment the line below to enable access logging */
```

**Comment by kazuho** [comment]
I believe that the issue has been fixed in #1602 thanks to the @Varbin's patch in the comment above.
